draft-ietf-ppm-dap.md

```diff
@@ -3077,60 +3271,67 @@ where `batch_interval` is the batch interval requested by the Collector.
-  minimum batch duration)
+~~~ tls-presentation
+struct {
+  start = 1729629000,
+  duration = 1000,
+} Interval
+~~~
 
-* both `batch_interval.start` and `batch_interval.duration` are divisible by
-  `time_precision`
+The `Query` received by the Leader or `BatchSelector` received by the Helper
+determines a valid set of batch bucket identifiers if
+`batch_interval.duration >= time_precision`.
 
 A batch consists of a sequence of contiguous batch buckets. That is, the set of
-batch buckets identifiers for the batch interval is
-`(batch_interval.start,
-  batch_interval.start + time_precision)`,
-`(batch_interval.start + time_precision,
-  batch_interval.start + 2*time_precision)`,
-  ...,
-`(batch_interval.start + batch_interval.duration - time_precision) +
-  batch_interval.start + batch_interval.duration)`.
+batch bucket identifiers for the batch interval is
+
+~~~ tls-presentation
+[
+  struct {
+    start = batch_interval.start,
+    duration = time_precision,
+  } Interval,
+  struct {
+    start = batch_interval.start + time_precision,
+    duration = time_precision,
+  } Interval,
+  ...
+  struct {
+    start = batch_interval.start + batch_interval.duration - time_precision,
+    duration = time_precision,
+  } Interval
+]
+~~~
 
 ## Leader-selected Batch Mode {#leader-selected-batch-mode}
 
-The leader-selected batch mode is used to support applications in which it is
-acceptable for reports to be batched in an arbitrary fashion by the Leader. Each
-batch of reports is identified by an opaque "batch ID". Both the reports
-included in each batch and the ID for each batch are allocated in an arbitrary
-fashion by the Leader.
+The leader-selected batch mode is used when it is acceptable for the Leader to
+arbitrarily batch reports. Each batch is identified by an opaque "batch ID"
+chosen by the Leader, which MUST be unique in the scope of the task.
+
+~~~ tls-presentation
+opaque BatchID[32];
+~~~
 
 The Collector will not know the set of batch IDs available for collection. To
-get the aggregate of a batch, the Collector issues a query, which does not
-include any information specifying a particular batch (see {{batch-modes}}). The
-Leader selects a recent batch to aggregate. The Leader MUST select a batch that
-has not yet been associated with a collection job.
+get the aggregate of a batch, the Collector issues a query for the next
+available batch. The Leader selects a recent batch to aggregate which MUST not
+yet have been associated with a collection job.
 
 The Aggregators can output batches of any size that is larger than or equal to
-`min_batch_size`. The target batch size, if any, is implementation-specific, and
-may be equal to or greater than the minimum batch size. Deciding how soon
-batches should be output is also implementation-specific. Exactly sizing batches
-may be challenging for Leader deployments in which multiple, independent nodes
-running the aggregate interaction (see {{aggregate-flow}}) need to be
-coordinated.
```

Comment:
Seems unhelpful to me to say that this part of the protocol is hard without saying what to do about it.

Reply:
we're saying that coordination in a massively distributed leader may make it impossible to hit the target batch size exactly.

Reply:
Fair enough. I put the original paragraph back.